File: youtube-links/src/shared/features/playlists/queries/GetAllUserPlaylists.ts
import { SortOrder } from "@/shared/abstractions/SortOrder";

export namespace GetAllUserPlaylists {

    export interface Query {
        page: number;
        pageSize: number;
        sortColumn: string;
        sortOrder: SortOrder;
        searchTerm: string;
        userId: number;
    }
}

File: YoutubeLinks.Shared/Features/Playlists/Queries/GetAllUserPlaylists.cs
using FluentValidation;
using MediatR;
using YoutubeLinks.Shared.Abstractions;
using YoutubeLinks.Shared.Features.Playlists.Responses;

namespace YoutubeLinks.Shared.Features.Playlists.Queries;

public static class GetAllUserPlaylists
{
    public class Query : QueryParameters, IRequest<PagedList<PlaylistDto>>
    {
        public int UserId { get; init; }
    }

    public class Validator : AbstractValidator<Query> { }
}